File: NeetCode/javascript/Hashing/0075. Sort Colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    let p=0
    let size=nums.length-1;
    for(let i=p;i<=size;i++){
        if(nums[i]===0)
        {
         [nums[i],nums[p]]=[[nums[p]],nums[i]];
         p++;
        }
        if(nums[i]===2)
        {
            [nums[i],nums[size]]=[nums[size],nums[i]];
            size--;
            i--;
        }
        

        
    }
};

File: NeetCode/C++/Hashing/0075. Sort Colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int p=0;
        int size=nums.size()-1;
        for(int i=p;i<=size;i++){
            if(nums[i]==0){
                swap(nums[i],nums[p]);
                p++;
            }
            if(nums[i]==2){
                swap(nums[i],nums[size]);
                size--;
                i--;
            }
        }
        
    }
};